weave/cache.py

```diff
@@ -113,11 +113,16 @@ def clear_cache() -> None:
 
 def get_user_cache_key() -> typing.Optional[str]:
     ctx = wandb_api.get_wandb_api_context()
+
     if ctx is None:
         if environment.is_public():
             raise errors.WeaveAccessDeniedError("No user set in public environment")
         return None
-    return ctx.user_id
+    if ctx.user_id:
+        return ctx.user_id
+    if ctx.cookies and ctx.cookies["anon_wandb"]:
```

Comment:
i think it would be cleaner if the weave cache logic did not have to know about the specific cookies / cookie name. This feels like we are coupling network implementation details with cache details. Is there a way that the middleware layer could store the anon_wandb cookie value in the context in such a way that it is clearly an anonymous session id? 

Note: these cookies should be considered "black box" inside the the weave application and are just forwarded to gorilla.